cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1120,6 +1120,8 @@ private ApplicationWithPrograms deployApp(NamespaceId namespaceId, @Nullable Str
   private void deleteApp(ApplicationId appId, ApplicationSpecification spec) throws IOException {
     //Delete the schedules
     scheduler.deleteSchedules(appId);
+    //Delete the schedules with -SNAPSHOT version
+    scheduler.deleteSchedules(new ApplicationId(appId.getNamespace(), appId.getApplication()));
```

Comment:
@jingtaowu3 can you confirm that we also need to delete the -SNAPSHOT schedules when do a deleteApp?

Reply:
so I think afaik from our previous discussion, it doesn't matter what appId we pass here? @jingtaowu3

Reply:
correct, version in appId will be ignored

Reply:
If that's the case then I should not need to add this extra delete.

Reply:
but the tests fail right?

Reply:
right, the tests suggested schedules were not deleted completely

Reply:
Okay created this for now: https://cdap.atlassian.net/browse/CDAP-19933, let us remove this code change however, since this needs more clarification

Reply:
reverted